skyportal/utils/offset.py

```diff
@@ -156,6 +157,101 @@ def get_ztfref_url(ra, dec, imsize, *args, **kwargs):
 }
 
 
+def calculate_best_position(
+    photometry, fallback=(None, None), how="snr", max_offset=0.5, sigma_clip=4.0
+):
+    """Calculates the best position for a source from its photometric
+       points
+
+    Parameters
+    ----------
+    photometry : list
+        List of Photometry objects
+    fallback : tuple, optional
+        The position to use if something goes wrong here
+    how : str
+        how to weight positional data:
+          snr = use the signal to noise
+          err = use the photometric uncertainty
+    max_offset : float, optional
+        How many arcseconds away should we ignore discrepant points?
+    sigma_clip : float, optional
+        Remove positions that are this number of std away from the median
+    """
+
+    if not isinstance(photometry, list):
+        log(
+            "Warning: No photometry given. Falling back to "
+            " original source position."
+        )
+        return fallback
+
+    # convert the photometry into a dataframe
+    phot_s = "[" + ",".join([str(x) for x in photometry]) + "]"
+    df = pd.DataFrame(json.loads(phot_s))
+
+    # remove limit data (non-detections)
+    try:
+        df = df[(df["flux"].notnull()) & (df["fluxerr"].notnull())]
+    except KeyError:
+        log(
+            "Photometry does not include fluxes. Falling back to "
+            " original source position."
+        )
+        return fallback
+
+    # remove observations with distances more than max_offset away
+    # from the median
+    try:
+        med_ra, med_dec = np.median(df["ra"]), np.median(df["dec"])
+    except TypeError:
+        log(
+            "Warning: could not find the median of the positions"
+            " from the photometry data associated with this source "
+        )
+        return fallback
+
+    # check to make sure that the median isn't too far away from the
+    # discovery position
+    if fallback != (None, None):
+        c1 = SkyCoord(med_ra * u.deg, med_dec * u.deg, frame='icrs')
```

Comment:
Why not use `SkyCoord` elsewhere for consistency?

Reply:
Unlike the other calculations could be a huge angle, if there is a catastrophic error in the photometry relative to the fiducial -- so I need the great circle distance here.